rotkehlchen/chain/evm/decoding/velodrome/decoder.py

```diff
@@ -256,3 +276,217 @@ def _decode_gauge_events(self, context: DecoderContext) -> DecodingOutput:
+                event.amount == ONE
+            ):
+                event.counterparty = self.counterparty
+                event.event_subtype = HistoryEventSubType.NONE
+                event.event_type = HistoryEventType.INFORMATIONAL
+                event.notes = f'Burn veNFT-{token_id} to unlock {amount} {self.token_symbol} from vote escrow'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == amount and
+                    event.address == self.voting_escrow_address
+            ):
+                event.counterparty = self.counterparty
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.notes = f'Receive {amount} {self.token_symbol} from vote escrow after burning veNFT-{token_id}'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_create_lock_event(self, context: DecoderContext) -> DecodingOutput:
+        in_event, out_event = None, None
+        token_id = int.from_bytes(context.tx_log.topics[2])
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        lock_time = deserialize_timestamp(int.from_bytes(context.tx_log.data[32:64]))
+
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.address == ZERO_ADDRESS and
+                    event.amount == ONE
+            ):
+                event.notes = f'Receive veNFT-{token_id} for locking {amount} {self.token_symbol} in vote escrow'  # noqa: E501
+                event.event_type = HistoryEventType.INFORMATIONAL
+                event.event_subtype = HistoryEventSubType.NONE
+                event.counterparty = self.counterparty
+                in_event = event
+
+            elif (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.address == self.voting_escrow_address and
+                    event.amount == amount
+            ):
+                event.notes = f'Lock {amount} {self.token_symbol} in vote escrow until {timestamp_to_date(lock_time, formatstr="%d/%m/%Y")}'  # noqa: E501
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.extra_data = {
+                    'token_id': token_id,
+                    'lock_time': lock_time,
+                }
+                event.event_type = HistoryEventType.DEPOSIT
+                event.counterparty = self.counterparty
+                out_event = event
```

Comment:
Also you might want to use the product field to set it to gauge. I believe you will need it in balances